src/plugins/irc-events/message.js

```diff
@@ -41,13 +42,23 @@ module.exports = function(irc, network) {
 		let from;
 		let highlight = false;
 		let showInActive = false;
+		let shouldIgnore = false;
 		const self = data.nick === irc.user.nick;
 
+		// Check if the sender is in our ignore list
+		shouldIgnore = network.ignoreList.some(function(entry) {
+			return Helper.compareHostmask(entry, data);
+		});
+
 		// Server messages go to server window, no questions asked
 		if (data.from_server) {
 			chan = network.channels[0];
 			from = chan.getUser(data.nick);
 		} else {
+			if (shouldIgnore) {
+				return;
+			}
```

Comment:
So messages from ignored people are trashed on reception, correct?

I explained that differently [here](https://github.com/thelounge/thelounge/issues/464#issuecomment-373262236), but I think we should not completely delete based ignores on reception. When ignoring someone, I think all their previous messages should be removed too, and when unignoring someone, all messages should re-appear.
Thoughts?

Reply:
I agree. Messages shouldn't disappear into the aether. They should be set as ignored somehow on the message and things that ignore them (highlighting, displaying) should use that feature.

Reply:
That's gonna be much more painful to implement what you are suggesting.

Reply:
Okay, not wanting to use a fallacy, but every other IRC client just ignores the messages after the mask has been added to the list. And when you remove the mask from the list, previous messages are not shown.

I think there's no easy way to do what you guys are asking for, pretty sure it'd need to make the client aware of these messages and just apply a CSS filter to them or something like that.

Reply:
However painful to implement, it's the better UX. I wouldn't say the first iteration of ignore support needs to work that way, since several other IRC clients I've used just irrevocably discard ignored messages. But it's a good goal to have `/ignore` and `/unignore` work retroactively.

Reply:
Yeah, having chatted with @xPaw on IRC, I agree that it would be fine to have this for the first version, but we want to work towawrds it after this PR.

@astorije ?

Reply:
That's fair. Your PR has a lot of useful work around managing ignores, we can improve the actual ignoring later (assuming we will though).

I don't think there needs to be CSS involved though: when a message is received, it is instantiated and added to the channel, but not sent to the client. When a client loads channels, messages are compared to the ignored list and those that should be ignored are not sent to the client.

Of course, it's more complex that just dropping ignored messages, as there will be more edge cases (dealing with messages that usually open a new query window, making this work without reloading the page, etc.) but that can be improved over time :)

Reply:
It would be nice if it worked similarly to weechat's filters. You can toggle those on and off and all matching messages are hidden/shown depending on the state of the filter.